### PR TITLE
Bump pinned Dendrite and SoftNPU artifacts

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

File: tools/ci_download_softnpu_machinery [hunk 1]
```diff
@@ -19,14 +19,14 @@ fi
...
+    curl -OL https://buildomat.eng.oxide.computer/wg/0/artefact/01GYB130ZXZKREJ0SSDXQXTP0F/DZ4rAowUPD7Ir3ZOdHwTDnsk7dtgSgQdRHoiaXJoHdqF29Xo/01GYB13H3Q4P7VQQV83B8TRAPT/01GYB1GET1WH9Y3566636EB27M/scadm
     chmod +x scadm
     mv scadm out/softnpu/
 fi
 
 # Fetch the pre-compiled sidecar_lite p4 program
 if [[ ! -f out/softnpu/libsidecar_lite.so ]]; then
     echo "fetching libsidecar_lite.so"
-    curl -OL https://buildomat.eng.oxide.computer/wg/0/artefact/01GTD3Y38K0Q14F2989R20QCBA/Ny1N0clhEwz2nfbsdTY8ta7pd9IYy1ofxG6ViCDN6Uy4xW3F/01GTD3YZ0TC9SCB4TFNYWP0DEM/01GTD4CA8KRGR8HK4S6B0DEEJC/libsidecar_lite.so
+    curl -OL https://buildomat.eng.oxide.computer/wg/0/artefact/01GYB130ZXZKREJ0SSDXQXTP0F/DZ4rAowUPD7Ir3ZOdHwTDnsk7dtgSgQdRHoiaXJoHdqF29Xo/01GYB13H3Q4P7VQQV83B8TRAPT/01GYB1GDMNCWD3K5MYXP203JYD/libsidecar_lite.so
```
Comment (luqmana) [inline tools/ci_download_softnpu_machinery:30]: I thought ry had added publishing to the `https://buildomat.eng.oxide.computer/public/file/oxidecomputer/...` style links that included the commit?

Comment (rcgoodfellow) [reply tools/ci_download_softnpu_machinery:30]: Yup, published artifacts [should now be available](https://github.com/oxidecomputer/sidecar-lite/blob/main/.github/buildomat/jobs/build-and-test.sh#L12-L30). We'll want the ones from the `release` series.

Comment (bnaecker) [reply tools/ci_download_softnpu_machinery:30]: I didn't see that, thanks! Will update now.

Comment (bnaecker) [reply tools/ci_download_softnpu_machinery]: Updated in d37f38469

**Review by luqmana — APPROVED** [review]

**Pull request merged**